xds/src/main/java/com/linecorp/armeria/xds/XdsResponseHandler.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.xds;
+
+import com.google.protobuf.Message;
+
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+
+/**
+ * Handles callbacks for {@link SotwXdsStream}.
+ * Note that it is important that exceptions are not raised from the callback.
+ * Otherwise, the infinite loop will break as a discovery request is not sent.
+ */
+interface XdsResponseHandler {
+
+    <T extends Message> void handleResponse(
+            ResourceParser<T> resourceParser, DiscoveryResponse value, SotwXdsStream sender);
```

Comment:
Is `resourceParser` necessary? `DiscoveryResponse` has the type in it so we can use the type to get the corresponding `ResourceParser`.

Reply:
Do you mean there is no need to pass the `resourceParser` and we can re-run `XdsResourceParserUtil.fromTypeUrl`?

We could, but what would be the advantage of doing so?

Reply:
In my mind, the `Stream` layer will do all the heavy lifting to ensure communication with the control plane is maintained and up-to-date.
For this reason I would rather keep `nonceMap`, `versionsMap` in the `XdsStream` layer.

https://github.com/line/armeria/blob/bfd4d90534b7437ba6c1d9d82e6077d710123cc0/xds/src/main/java/com/linecorp/armeria/xds/SotwXdsStream.java#L214

In order to do so, I think it makes sense that the `ResourceType` (or `ResourceParser`) is determined inside `SotwXdsStream`. I didn't see a need to re-compute this so I just passed it via parameter. I'd rather keep this the way it is, but let me know if you feel strongly about this 😄

Reply:
Understood. That makes sense. 👍